OpenNefia.Content/Feats/FeatsSystem.cs

```diff
@@ -0,0 +1,42 @@
+﻿using OpenNefia.Core.GameObjects;
+using OpenNefia.Core.Prototypes;
+using OpenNefia.Core.Stats;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.Feats
+{
+    public interface IFeatsSystem : IEntitySystem
+    {
+        int Level(FeatsComponent feats, PrototypeId<FeatPrototype> id);
+        Stat<int> Ensure(FeatsComponent feats, PrototypeId<FeatPrototype> id);
+        bool TryGetKnown(FeatsComponent feats, PrototypeId<FeatPrototype> protoId, [NotNullWhen(true)] out Stat<int>? stat);
+    }
+    public sealed partial class FeatsSystem : EntitySystem, IFeatsSystem
+    {
+        public Stat<int> Ensure(FeatsComponent feats, PrototypeId<FeatPrototype> id)
```

Comment:
I feel like `Ensure`-type functions should allocate the entry if it's missing like with `EntityManager.EnsureComponent<>()`. Is there any place this is used where it wouldn't be impractical to use `Level()` instead?

Reply:
true, removed it